Code_Exercises/In_Order_Queue/solution1.cpp

```diff
@@ -0,0 +1,101 @@
+#include <sycl.hpp>
+
+#define MAD_4(x, y)                                                                                \
+  x = y * x + y;                                                                                   \
+  y = x * y + x;                                                                                   \
+  x = y * x + y;                                                                                   \
+  y = x * y + x;
+#define MAD_16(x, y)                                                                               \
+  MAD_4(x, y);                                                                                     \
+  MAD_4(x, y);                                                                                     \
+  MAD_4(x, y);                                                                                     \
+  MAD_4(x, y);
+#define MAD_64(x, y)                                                                               \
+  MAD_16(x, y);                                                                                    \
+  MAD_16(x, y);                                                                                    \
+  MAD_16(x, y);                                                                                    \
+  MAD_16(x, y);
+
+template <class T> static T busy_wait(size_t N, T i) {
```

Comment:
I would avoid using `wait` in a func name just so that the user doesn't associate these with `wait` member functions for SYCL classes. Maybe better to call this `do_work`

Reply:
Good point. I will call it `sleep`, as it is really what this thing is doing... Burning some watt :)